feat: Adds Input-iban country blacklisting

andrei1902 opened the pull request:
## What I did

1. Added IsBlacklistedCountryIBAN class on validators (error messages are the same as IsCountryIBAN validator)
2. Added unit test for new validator

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ing-bank/lion?pullRequest=1063) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ing-bank/lion?pullRequest=1063) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ing-bank/lion?pullRequest=1063) it.</sub>

gerjanvangeest commented:
Hi @andrei1902,

Thank you for the new feature, but i have some questions.

1. A blacklisted validator would mean in my opinion that IBANs from that country should not be used, but the validation feedback message says I should use (in this case) a Romanian IBAN.
2. Why not use an array to define multiple countries at once (could make the default Validation message more complex)?

andrei1902 commented:
> Hi @andrei1902,
> 
> Thank you for the new feature, but i have some questions.
> 
> 1. A blacklisted validator would mean in my opinion that IBANs from that country should not be used, but the validation feedback message says I should use (in this case) a Romanian IBAN.
> 2. Why not use an array to define multiple countries at once (could make the default Validation message more complex)?

1. I added translations for this particular validator to keep it in step with the IsCountryIBAN validator.
2. Correct! The usage of an array of countries would complicate the error message. I updated the README.md to show multiple validator usage.

jorenbroekema commented on packages/input-iban/README.md:
```diff
@@ -90,3 +90,23 @@ export const countryRestrictions = () => html`
+By default, we validate the input to ensure the IBAN is valid.
+To get the default feedback message for this default validator, use `loadDefaultFeedbackMessages` from `@lion/form-core`.
+
+In the example below, we show how to use an additional validator that restricts the `input-iban` to IBANs from certain countries.
```
```suggestion
In the example below, we show how to use an additional validator that blocks IBANs from certain countries.
```

jorenbroekema reviewed (changes requested):
I would suggest changing the name from IsBlacklistedCountryIBAN to `IsNotCountryIBAN` to make it consistent with `IsCountryIBAN` and to make clear that it is the opposite behaviour. Other than that, I think it looks good 👍 .

We can look some other time into making both of them accept an array perhaps, but as you mentioned it will complicate error messages, so might be a tricky improvement.

andrei1902 commented:
I pushed the changes you suggested.

jorenbroekema commented on packages/input-iban/README.md:
```diff
@@ -90,3 +90,23 @@ export const countryRestrictions = () => html`
+    label="IBAN"
+  ></lion-input-iban>
+  <br />
+  <small>Demo instructions: you can use RO 89 RZBR 6997 3728 4864 5577</small>
```
Maybe a little confusing :P since RO is blacklisted, but you want to show off this validation by putting an invalid country.

```suggestion
  <small>Demo instructions: Try <code>RO 89 RZBR 6997 3728 4864 5577</code> and watch it fail</small>
```

jorenbroekema commented on packages/input-iban/src/validators.js:
```diff
@@ -153,3 +153,39 @@ export class IsCountryIBAN extends IsIBAN {
+   */
+  static async getMessage(data) {
+    await loadTranslations();
+    return localize.msg('lion-validate+iban:error.IsBlaklistedCountryIBAN', data);
```
I would also change the name of the validation messages, makes it easier to find if it's the same name as the validator. 
```suggestion
    return localize.msg('lion-validate+iban:error.IsNotCountryIBAN', data);
```

jorenbroekema commented on packages/input-iban/translations/bg.js:
```diff
@@ -16,5 +16,20 @@ export default {
       'RO {Румънски}\n' +
       'other {{params}}\n' +
       '} {fieldName}.',
+    IsBlaklistedCountryIBAN:
```
Same comment as above

jorenbroekema commented on packages/input-iban/test/validators.test.js:
```diff
@@ -19,4 +19,14 @@ describe('IBAN validation', () => {
+    expect(deValidator.execute('DE89370400440532013000')).to.be.true;
+    expect(nlValidator.execute('DE89370400440532013000')).to.be.false;
+    expect(deValidator.execute('NL17INGB0002822608')).to.be.false;
+    expect(nlValidator.execute('foo')).to.be.false;
```
I think test is failing on this line. Which is expected. The IsNotCountryIBAN does not fail on "foo", but the IsIBAN validator that is applied by default on input-iban, does fail. I think this line can be removed, since it's not a test for the IsNotCountryIBAN, but more of a input-iban/IsIBAN specific test.

jorenbroekema commented:
Hey @andrei1902  I added some final comments, just small things. After that, you just need to sign the CLA and then I'm fine to merge it 👍  thanks for your work so far on contributing this validator!

jorenbroekema commented:
Closing in favor of https://github.com/ing-bank/lion/pull/1120 , made some general improvements to these validators there and added array support for this one, and changed the translation key as suggested to IsNotCountryIBAN.
